src/arrayMethodSort.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let element = 0; element < this.length - i - 1; element++) {
+        if (compareFunction && typeof compareFunction === 'function') {
```

Comment:
instead of this strange logic just add the default value for compareFunction argument